source/MRMesh/MRFillContours2D.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+#include "MRMeshFwd.h"
+
+namespace MR
+{
+
+/**
+ * @brief fill holes with border in same plane (i.e. after cut by plane)
+ * @param mesh - mesh with holes
+ * @param holesEdges - edges set represents holes borders that should be filled
+ * should be not empty
+ * edges should have invalid left face (FaceId == -1)
+ * @return true if holes filled, otherwise - false
+ */
+MRMESH_API bool fillContours2D( Mesh& mesh, const EdgePath& holesEdges );
```

Comment:
EdgePath is typically a number of connected edges, where edge[i].dest == edge[i+1].org, but here you pass not edge path, but an representative edge of each hole. See example:
```
    // returns one edge with no valid left face for every boundary in the mesh
    MRMESH_API std::vector<EdgeId> MeshTopology::findHoleRepresentiveEdges() const;
```
So please rename the parameter and make its type std::vector<EdgeId>

Reply:
fixed